trulens_eval/trulens_eval/tru.py

```diff
@@ -586,6 +589,11 @@ def listen_to_dashboard(proc: subprocess.Popen, pipe, out, started):
             target=listen_to_dashboard,
             args=(proc, proc.stderr, out_stderr, started)
         )
+
+        # Purposely block main process from ending and wait for dashboard. 
+        Tru.dashboard_listener_stdout.daemon=False
```

Comment:
Why are we adding the wait for dashboard to the test? Are we worried about dashboard breakages?

Reply:
no; this is the existing functionality. Just being explicit here. 
Also; this is not the test; this is the main branch. all threading changes are actual usage.

The main thing here is currently if you call a script with the context managers; the script will never end (This is true also for pytest which is why its in this PR)

However; if you run a script with tru.run_dashboard, without this; the script will end and the dashboard will be be killed with the running script